### DISCUSSION TITLE
What is Sn (Power Rating) and why is so often set to 100?

### DISCUSSION THREAD

**Discussion by MaKla89** [CURENT/andes#447] [Q&A] [answered]
Cheers, this will be a quick and simple question! :) 

I've now used several pre-made cases and am now trying to create my own first (very small) test-system. But I struggle with getting the initial values right and I think that it has to do with the Power rating. In the documentation, it is mostly labeled as unit-less (Slack, PV, ...) and in some cases the unit is MVA (SynGen) or MW (Line). In many many examples the Sn is simply 100, which looks like a percentage of its nominal designed capacity (which is then calculated based on other inputs?).  I did not find any explicit part in the documentation about Sn (and whether it is always the same or if Sn might mean something different in different models).

I only know power rating as an absolute number, typically in MW / MVA, but the values found in some examples make me think that this understanding of mine is wrong. Is it a value that needs to be converted to p.u.? For example the ieee14_full.xlsx: ALL "power rating"  for PV, Slack, Line and GENROU are set to 100. This confuses me, if at least for some models they are supposed to be the actual nominal capacity of the asset :D

**Comment by cuihantao** [comment]
Sn is the power rating of the device. Many device parameters are given in per unit in the device Sn. The default power base for the system is 100 MVA.

Many of the generators in some cases have Sn as 100 so that the parameters under the device base is the same as the system base.


Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: MaKla89 ***@***.***>
Sent: Wednesday, April 26, 2023 6:21:57 AM
To: CURENT/andes ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [CURENT/andes] What is Sn (Power Rating) and why is so often set to 100? (Discussion #447)


Cheers, this will be a quick and simple question! :)

I've now used several pre-made cases and am now trying to create my own first (very small) test-system. But I struggle with getting the initial values right and I think that it has to do with the Power rating. In the documentation, it is mostly labeled as unit-less (Slack, PV, ...) and in some cases the unit is MVA (SynGen) or MW (Line). In many many examples the Sn is simply 100, which looks like a percentage of its nominal designed capacity (which is then calculated based on other inputs?). I did not find any explicit part in the documentation about Sn (and whether it is always the same or if Sn might mean something different in different models).

I only know power rating as an absolute number, typically in MW / MVA, but the values found in some examples make me think that this understanding of mine is wrong. Is it a value that needs to be converted to p.u.? For example the ieee14_full.xlsx: ALL "power rating" for PV, Slack, Line and GENROU are set to 100. This confuses me, if at least for some models they are supposed to be the actual nominal capacity of the asset :D

**Comment by cuihantao** [comment]
`Sn` in the power flow devices, including PV, PQ, and Slack, are not used for any calculation. It is because the `p0` and `q0` of PQ, for example, are specified as per unit in the system base. Technically, it can be implemented the other way around (namely, having each device specify `Sn` and let `p0` be a per unit in the `Sn` base). 

The initialization issue may have to do with the `Sn` of the dynamic device. In a small Sn, the per unit quantities are large and will probably hit some limits.

**Reply by MaKla89** [reply]
Thanks for these clarifications! Lines would then be "power flow devices" and ignore the Sn-parameter, too? :)

**Comment by jinningwang** [comment, marked as answer]
If you're looking for more information about the per-unit system, you can check out this resource: https://docs.andes.app/en/latest/getting_started/misc.html?highlight=per%20unit#per-unit-system.

**Reply by MaKla89** [reply]
This is a very good page which might be placed a bit too un-prominent (Also, I could not find it when searching for "Power rating").

I am still a bit unsure how to handle some of the values and conversions, but that is mostly on me :) 

As an idea: Would it be good to have a simple "conversion-example"? I know, this is not really the responsibility of ANDES, but it might be fairly simple to set up and clarify many questions.

**Comment by jinningwang** [comment]
In ANDES, per unit conversion is applied to ``TDS`` devices, but not ``PFlow`` devices. A ``PFlow`` devices is a device that has ``self.flags.pflow = True`` in the model definition. Similarly, a ``TDS`` device is a device that has ``self.flags.tds = True`` in the model definition.

Here is an example to further clarify the per-unit conversion in ANDES, [per_unit.pdf](https://github.com/CURENT/andes/files/11346304/per_unit.pdf).

@cuihantao Hantao, is it helpful to include this notebook file in the ``examples/demonstration``? I can do a PR if necessary.

**Reply by MaKla89** [reply]
This is a nice document, thanks a lot :) Sadly, that very example brought me to another (equally simple) question:

If for TDS-devices (like the GENROU) , pu-conversion is done automatically, do I then need to enter absolute values to all TDS-devices in the xlsx-file? 

In other words: Your example NICELY explains how ANDES converts internally, but I'm still somewhat confused about what kind of input is expected of me. If pu-conversion is done internally, then I'd assume I'd have to enter everything in nominal values (At least for TDS-enabled assets). For assets that have both the TDS and PF-tag "True", I'd then not know whether its' nominal values or manually converter p.u. -values, etc. I simply obey to the model reference pages, assume system reference power as 100 MVA (usually), the local voltage level of the asset as the reference voltage and so on. For lines I enter r, x and b values manually converted into p.u. based on the base impedance. 

Is there ANY case where the p.u. values are RE-based if the assets 'Sn' deviates from the system power rating?  (Do I use the system power rating 100 MVA for r, x and b in lines or the LINE power rating 'Sn' ).

But to be honest: If every other user manages this without pestering you, then I will simply accept a knot in my head and stop asking :D

**Reply by cuihantao** [reply]
Here's the rule set:
1. Parameters whose "properties" column contain conversion flags (such as `power`, `z`, `y`) should be PROVIDED as per unit values in the bases (`Sn` and `Vn`, if applicable) of the device. The parameters are INTERNALLY re-based to the system base.
2. Otherwise, see the "description" and "unit" for specifying the parameter. These parameters are used AS IS, not re-based.

Below are two examples: 

PQ model for power flow:
 
- Check https://docs.andes.app/en/latest/groupdoc/StaticLoad.html#pq for how `p0` and `q0` should be specified
- "Properties" of `p0` and `q0` do not contain any conversion flag, so rule 2 applies.
- The description says that `p0` and `q0` are active and reactive power in system base.

GENROU model
Determine how a generator with a capacity of 200 MVA and H=6 s (in the machine base) should be entered. Manufacturere usually provide parameters in per unit in the device base.

- Check https://docs.andes.app/en/latest/groupdoc/SynGen.html#gencls
- 200 MVA goes to `Sn`
- `M=2H` is the machine starting time constant. `M` is expected to be per unit values in the machine base. So the input to `M` is simply `2*6 = 12`.

As you can see, these manufacturer parameters (i.e., parameters in the device base) can be entered as is to an ANDES parameter that has a conversion flag.

**Reply by jinningwang** [reply]
You can specify the ``Sn`` in the in the TDS-devices if it has a different device base. Check out the built-in case ``npcc.xlsx``, and you can follow the same style of it.
> If for TDS-devices (like the GENROU) , pu-conversion is done automatically, do I then need to enter absolute values to all TDS-devices in the xlsx-file?

In the [NumParam definition](https://docs.andes.app/en/latest/modeling/_generated/andes.core.param.NumParam.html?highlight=ipower), the following parameters are used to determine what type of conversion is applied: ``power: bool = False, ipower: bool = False, voltage: bool = False, current: bool = False, z: bool = False, y: bool = False, r: bool = False, g: bool = False``. In the [GENROU Parameters](https://docs.andes.app/en/latest/groupdoc/SynGen.html#id4), you can find out there is a column ``Properties`` to indicate this.
> For assets that have both the TDS and PF-tag "True", I'd then not know whether its' nominal values or manually converter p.u. -values, etc. I simply obey to the model reference pages, assume system reference power as 100 MVA (usually), the local voltage level of the asset as the reference voltage and so on. For lines I enter r, x and b values manually converted into p.u. based on the base impedance.
